doc/reference/reference_lua/msgpack.rst

```diff
@@ -526,3 +564,210 @@ and :ref:`YAML <yaml-cfg>`.
+        -- Create a MsgPack object from a Lua object of any type
+        mp = msgpack.object(123)
+        mp = msgpack.object("foobar")
+        mp = msgpack.object({1, 2, 3})
+        mp = msgpack.object({foo = 1, bar = 2})
+        mp = msgpack.object(box.tuple.new(1, 2, 3))
+
+..  _msgpack-object-from-raw:
+
+..  function:: object_from_raw(msgpack_string)
+
+    Since version 2.10.1.
+
+    Create a MsgPack object from a raw MsgPack string.
+
+    :param string msgpack_string: a raw MsgPack string.
+
+    :return: a MsgPack object
+
+    :rtype: userdata
+
+    **Example:**
+
+    ..  code-block:: lua
+
+        local msgpack = require('msgpack')
+        local data = msgpack.encode({1, 2, 3})
+        local mp = msgpack.object_from_raw(data)
+
+..  _msgpack-object-from-raw-pointer:
+
+..  function:: object_from_raw(C_style_string_pointer, size)
+
+    Since version 2.10.1.
+
+    Create a MsgPack object from a raw MsgPack string. The address of the MsgPack string is supplied as a C-style string pointer
+    such as the ``rpos`` pointer inside an ``ibuf`` that the :ref:`buffer.ibuf() <buffer-ibuf>` creates.
+    A C-style string pointer may be described as ``cdata<char *>`` or ``cdata<const char *>``.
+
+    :param buffer C_style_string_pointer: a pointer to a raw MsgPack string.
+    :param integer size: number of bytes in the raw MsgPack string.
+
+    :return: a MsgPack object
+
+    :rtype: userdata
+
+    **Example:**
+
+    ..  code-block:: lua
+
+        local msgpack = require('msgpack')
+        local buffer = require('buffer')
+        local buf = buffer.ibuf()
+        msgpack.encode({1, 2, 3}, buf)
+        local mp = msgpack.object_from_raw(buf.buf, buf:size())
+
+..  _msgpack-is-object:
+
+..  function:: is_object(some_argument)
+
+    Since version 2.10.1.
+
+    Check if the given argument is a MsgPack object.
+
+    :param some_agrument: any argument.
+
+    :return: ``true`` if the argument is a MsgPack object; otherwise, ``false``
+
+    :rtype: boolean
+
+    **Example:**
+
+    ..  code-block:: lua
+
+        local msgpack = require('msgpack')
+        local mp = msgpack.object(123)
+        msgpack.is_object(mp) -- returns true
+        msgpack.is_object({}) -- returns false
+
+..  _msgpack-object-methods:
+
+..  class:: msgpack_object
+
+    The MsgPack object has the following methods:
+
+    ..  method:: decode()
+
+        Since version 2.10.1.
+
+        Decode MsgPack data in a MsgPack object and return a Lua object.
+
+        :return: a Lua object
+
+        :rtype: Lua object
+
+    ..  method:: iterator()
+
+        Since version 2.10.1.
+
+        Return an iterator over the MsgPack data.
+
+        A MsgPack iterator object has its own :ref:`set of methods <msgpack-object-iterator-methods>`.
+
+        :return: an iterator object over the MsgPack data
+
+        :rtype: userdata
+
+    A MsgPack object can be passed to the MsgPack encoder with the same effect as passing the original Lua object:
```

Comment:
Maybe, move it under `class:: msgpack_object`? Currently, visually it looks like this belongs to `method:: iterator()`.